packages/eos-components/src/components/mixins/columnsMixin.js

```diff
@@ -0,0 +1,34 @@
+import { sm, md } from '../../styles.scss';
+
+export default {
+  data() {
+    return {
+      width: 0,
+    };
+  },
+  computed: {
+    columns() {
+      if (this.width < Number(sm)) {
+        return 1;
+      }
+
+      if (this.width <= Number(md)) {
+        return 2;
+      }
+
+      return 3;
+    },
+  },
+  created() {
+    this.width = window.innerWidth;
+    window.addEventListener("resize", this.onResize);
```

Comment:
This is the same way we implemented it for the chromeos demo. Since then I learnt about [matchMedia()](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) which is more performant because it triggers the callback when the breakpoint changes. See also [this bootstrap-vue discussion](https://github.com/bootstrap-vue/bootstrap-vue/issues/5284#issuecomment-622392181).

Reply:
Oh, thanks for the links, I didn't know about the `matchMedia` and that's exactly what we need here.